bindings/rust/bench/src/harness.rs

```diff
@@ -14,27 +14,37 @@ pub fn read_to_bytes(path: &str) -> Vec<u8> {
     read_to_string(path).unwrap().into_bytes()
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
 pub enum Mode {
     Client,
     Server,
 }
 
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq)]
+pub enum HandshakeType {
```

Comment:
nit: I don't like boolean, naming. Perhaps `MutualAuth` and `ServerAuth`? TLS 1.2 actually offered `NoAuth` but looks like that has been deprecated for TLS 1.3: https://www.rfc-editor.org/rfc/rfc8446#appendix-C.5

Reply:
Rustls has `NoClientAuth` and `AllowAnyAuthenticatedClient`, while s2n-tls has `CERT_AUTH_NONE` and `CERT_AUTH_REQUIRED`, which aren't directly boolean names. However, I think just `ServerAuth` doesn't indicate strongly enough that there is no client authentication, i.e. with mutual authentication there's still server authentication.

Reply:
I like MutualAuth and ServerAuth. That gets the point across. Client authentication really not that common, most people will know server auth is on by default and client auth is off by default.